File: frontend/src/api/dto/teamInvitationListResponse.ts
import type { TeamInvitationDto } from './teamInvitationDto.ts'

/**
 * Paginated list of a team's invitations
 */
export interface TeamInvitationListResponse {
  /** Invitations */
  invitations: TeamInvitationDto[]
  /** Total number of invitations */
  total: number
  /** Current page (0-indexed) */
  page: number
  /** Page size */
  size: number
}

File: backend/src/main/java/fr/pedalons/dto/teams/response/TeamInvitationListResponse.java
package fr.pedalons.dto.teams.response;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated list of a team's invitations")
@ValidateSchema
public record TeamInvitationListResponse(
    @Schema(description = "Invitations", required = true) List<TeamInvitationDto> invitations,
    @Schema(description = "Total number of invitations", required = true) long total,
    @Schema(description = "Current page (0-indexed)", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}
